tgui/packages/tgui/interfaces/CameraConsole.js

```diff
@@ -0,0 +1,138 @@
+import { filter, sortBy } from 'common/collections';
+import { flow } from 'common/fp';
+import { classes } from 'common/react';
+import { createSearch } from 'common/string';
+import { Fragment } from 'inferno';
+import { useBackend, useLocalState } from '../backend';
+import { Button, ByondUi, Input, Section } from '../components';
+import { refocusLayout, Window } from '../layouts';
+
+/**
+ * Returns previous and next camera names relative to the currently
+ * active camera.
+ */
+const prevNextCamera = (cameras, activeCamera) => {
+  if (!activeCamera) {
+    return [];
+  }
+  const index = cameras.findIndex(camera => (
+    camera.name === activeCamera.name
+  ));
+  return [
+    cameras[index - 1]?.name,
+    cameras[index + 1]?.name,
+  ];
+};
+
+/**
+ * Camera selector.
+ *
+ * Filters cameras, applies search terms and sorts the alphabetically.
+ */
+const selectCameras = (cameras, searchText = '') => {
+  const testSearch = createSearch(searchText, camera => camera.name);
+  return flow([
+    // Null camera filter
+    filter(camera => camera?.name),
+    // Optional search term
+    searchText && filter(testSearch),
+    // Slightly expensive, but way better than sorting in BYOND
```

Comment:
Could pre-emptively sort the list, then you don't have to worry about it being out of order.

Reply:
Too complicated.